### ISSUE TITLE
reply gate additional feature - limit people who can reply to (only people following account) 

### ISSUE THREAD

**Issue by monicaellerose** [bluesky-social/social-app#2547]
I would like to be able to limit who can reply to posts by people who follow my account

i think this would help limit bad actors in the comments section. 

this is similar to how instagram does it

ideally this would have an option for both followed/following 
![image](https://github.com/bluesky-social/social-app/assets/66647238/1a20a8f3-5669-46ed-9c0e-738c29a13d5f)

![bafkreib6l6wr65qxdn434mdqao346sa2apayuwkmcz6qycowstvu2moyre](https://github.com/bluesky-social/social-app/assets/66647238/c6ec197c-87a9-4dac-a046-72b7847d0336)

**Comment by pfrazee** [comment]
@monicaellerose Always interested in improvements to the threadgate

There's currently no ability to control who follows you, so this would be easily circumventable by the person hitting follow. I think the friction is still actually pretty useful, but I just want to flag that and ensure you agree

Cheers!

**Comment by monicaellerose** [comment]
i

i def agree,  
anything that adds friction here i think is an improvement, it can go two ways "oh i thought i was already following this user, and oh this comment isnt worth the follow"
bad actors will always try to find ways to circumvent, but can still give pause to some of the more unhinged comments and i think thats a win.

> @monicaellerose Always interested in improvements to the threadgate
> 
> There's currently no ability to control who follows you, so this would be easily circumventable by the person hitting follow. I think the friction is still actually pretty useful, but I just want to flag that and ensure you agree
> 
> Cheers!

**Comment by monicaellerose** [comment]
adding the ability for people to comment if they have  custom domains - is another interesting take on this  https://bsky.app/profile/eva.computer/post/3klqedkjbuh2q

**Comment by monicaellerose** [comment]
throwing in - requiring a default pfp to comment bc why not.

**Comment by PSingletary** [comment]
> adding the ability for people to comment if they have  custom domains - is another interesting take on this  https://bsky.app/profile/eva.computer/post/3klqedkjbuh2q 

I can 100% get behind this

**Comment by monicaellerose** [comment]
> @monicaellerose Always interested in improvements to the threadgate
> 
> There's currently no ability to control who follows you, so this would be easily circumventable by the person hitting follow. I think the friction is still actually pretty useful, but I just want to flag that and ensure you agree
> 
> Cheers!

yeah i think as bluesky continues to evolve this is becoming more and more necessary
id also +1 the ability to add custom keywords to a reply gate, and certain labels  (to either approve or deny commenting), and posts with nsfw media,

**Comment by Tahvohck** [comment]
See also #2757 for a related idea for how to limit people.

@pfrazee 
> There's currently no ability to control who follows you, so this would be easily circumventable by the person hitting follow. I think the friction is still actually pretty useful, but I just want to flag that and ensure you agree

Since follow records are public, you could probably make it so that they have to have been following you for a certain amount of time before they can reply, similar to how Youtube chat can be set up. Either have that be configurable by the poster, or some suitable baked in default (though if it's going to be an added feature, it might as well be done with full effort and allow the time frame to be custom)

While this is under discussion, can someone with knowledge of the codebase point out where this would get implemented? I find an issue with open source contributions is figuring out just where those contributions would be made in the first place.

**Comment by mozzius** [comment]
This was added
